pkg/apis/monitoring/v1/prometheus_types.go

```diff
@@ -798,6 +798,17 @@ type PrometheusSpec struct {
 	// Maximum number of bytes used by the Prometheus data.
 	RetentionSize ByteSize `json:"retentionSize,omitempty"`
 
+	// ShardRetentionPolicy defines the retention policy for the Shard's statefulset.
+	// ShardRetentionPolicy is an experimental feature, only enabled when the feature gate
+	// `ShardRetentionPolicies` is enabled.
```

Comment:
We can also take inspiration from https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/#feature-stages and categorize gates as alpha, beta and GA.

```suggestion
	// (Alpha) Using this field requires the ShardRetentionPolicies feature gate to be enabled.
```

Reply:
Good idea :)